### PR TITLE
Fix Crash in MixedRealityKeyboard.cs by unregistering for event handlers

### PR DESCRIPTION
**Pull request by julenka**
## Overview
MixedRealityKeyboard was not unregistering event handlers for InputPane::Showing and InputPane::Hiding events, which could cause the application to crash when exiting, or if some other object tried to unsubscribe to InputPane::Hiding. Fix this by unsubscribing from the InputPane::Hiding event.

## Changes
- Fixes: #7739

### CODE REVIEW

File: Assets/MRTK/SDK/Experimental/MixedRealityKeyboard/MixedRealityKeyboardBase.cs [hunk 1]
```diff
@@ -104,22 +104,37 @@ private enum KeyboardState
 
         #region MonoBehaviour Implementation
 
-        /// <summary>
```
Comment (julenka) [inline Assets/MRTK/SDK/Experimental/MixedRealityKeyboard/MixedRealityKeyboardBase.cs:107]: Add back this summary comment.

**Comment by julenka** [comment]
@keveleigh  I tried to add @kevbarn as a reviewer but cannot add him to the code review. He helped me with this bug, wanted to get his feedback.

**Comment by keveleigh** [comment]
People have to be on a specific list to be requestable for review as a collaborator, whether internal or external. I added him!

File: Assets/MRTK/SDK/Experimental/MixedRealityKeyboard/MixedRealityKeyboardBase.cs [hunk 2]
```diff
@@ -142,7 +160,7 @@ private IEnumerator UpdateState()
                 yield return null;
             }
         }
-#endif
+#endif //!UNITY_EDITOR && UNITY_WSA
```
Comment (keveleigh) [inline Assets/MRTK/SDK/Experimental/MixedRealityKeyboard/MixedRealityKeyboardBase.cs:163]:
```suggestion
#endif // WINDOWS_UWP
```

File: Assets/MRTK/SDK/Experimental/MixedRealityKeyboard/MixedRealityKeyboardBase.cs [hunk 3]
```diff
@@ -104,22 +104,40 @@ private enum KeyboardState
 
         #region MonoBehaviour Implementation
 
+#if !UNITY_EDITOR && UNITY_WSA
```
Comment (keveleigh) [inline Assets/MRTK/SDK/Experimental/MixedRealityKeyboard/MixedRealityKeyboardBase.cs:107]: Are we able to [use `WINDOWS_UWP` here](https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/CodingGuidelines.html#use-platform-dependent-compilation-to-ensure-the-toolkit-wont-break-the-build-on-another-platform)?

```suggestion
#if WINDOWS_UWP
```

**Comment by keveleigh** [comment]
@julenka Should this go into 2.4 stabilization?

**Pull request merged**